src/symtable/SymbolTableScope.ts

```diff
@@ -120,27 +106,40 @@ export class SymbolTableScope {
     }
 
     lookup(name: string): Symbol_ {
-        return new Symbol_(name, this.symbols[name], this.__check_children(name), this.name === "top");
-        // let sym;
-        // if (!(name in this.symbols)) {
-        //     const flags = this.symFlags[name];
-        //     const namespaces = this.__check_children(name);
-        //     sym = this.symbols[name] = new Symbol_(name, flags, namespaces);
-        // } else {
-        //     sym = this.symbols[name];
-        // }
-        // return sym;
+        // return new Symbol_(name, this.symbols[name], this.checkChildren(name), this.name === "top");
+        if (this.symbolObjectCache.has(name)) {
+            return this.symbolObjectCache.get(name)!;
+        }
+
+        if (!(name in this.symbols)) {
+            throw new Error(`symbol ('${name}') not found!`);
+        }
+
+        this.symbolObjectCache.set(
+            name,
+            new Symbol_(name, this.symbols[name], this.checkChildren(name), this.name === "top")
+        );
+
+        return this.symbolObjectCache.get(name)!;
```

Comment:
```suggestion
        if (this.symbolObjectCache.has(name)) {
            return this.symbolObjectCache.get(name)!;
        }
        const flag = this.symbols[name];
        if (flag === undefined) {
            throw new Error(`symbol ('${name}') not found!`);
        }
        const symbol = new Symbol_(name, flag, this.checkChildren(name), this.name === "top");
        this.symbolObjectCache.set(name, symbol);
        return symbol;
```

mostly to avoid double lookups.